modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -132,10 +133,27 @@
 
       (isa? coercion :Coercion/YYYYMMDDHHMMSSString->Temporal)
       {"$dateFromString" {:dateString field-name
-                          :format "%Y%m%d%H%M%S"
-                          :onError field-name}}
+                          :format     "%Y%m%d%H%M%S"
+                          :onError    field-name}}
 
-      :else field-name)))
+      ;; mongo only supports datetime
+      (isa? coercion :Coercion/ISO8601->DateTime)
+      {"$dateFromString" {:dateString field-name
+                          :onError    field-name}}
+
+
+      (isa? coercion :Coercion/ISO8601->Date)
+      (throw (ex-info (tru "Driver Mongo does not support {0}" coercion)
+                      {:type              qp.error-type/unsupported-feature
+                       :coercion-strategy coercion}))
+
+
+      (isa? coercion :Coercion/ISO8601->Time)
```

Comment:
`$dateFromString` can accept an optional `:format` parameter so it seems like you should be able to parse times as well.

Reply:
The `$dateFromString` function requires a Date component.

I.e if we have a source column like `09:09:30` and try to parse it with `%H:%M%S` format, it'll give an error 

```sh
sample> db.nestedData.aggregate([{ $project: { date: { $dateFromString: { dateString: "$Info.started", format: "%H:%M:%S" } } } }])
MongoServerError: PlanExecutor error during aggregation :: caused by :: an incomplete date/time string has been found, with elements missing: "09:09:30"
```